pkg/kratos/service.go

```diff
@@ -387,6 +391,72 @@ func (s *Service) UpdateRecoveryFlow(
 	return returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) UpdateIdentifierFirstLoginFlow(
+	ctx context.Context, flow string, body kClient.UpdateLoginFlowBody, cookies []*http.Cookie,
+) (*BrowserLocationChangeRequired, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.UpdateIdentifierFirstLoginFlow")
+	defer span.End()
+
+	csrfToken, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetCsrfTokenOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing csrf token")
+	}
+
+	identifier, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetIdentifierOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing identifier")
+	}
+
+	form := url.Values{}
+	form.Set("csrf_token", *csrfToken)
+	form.Set("identifier", *identifier)
+	form.Set("method", "identifier_first")
+
+	u, err := url.Parse(s.kratosPublicURL + "/self-service/login")
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid kratos URL: %w", err)
+	}
+	q := u.Query()
+	q.Set("flow", flow)
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), strings.NewReader(form.Encode()))
+	if err != nil {
+		s.logger.Errorf("failed to create http request: %s", err)
+		return nil, nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
https://pkg.go.dev/github.com/ory/kratos-client-go#LoginFlow.SetType

based on what I read, if we set the type to API we won't need to perform form posts anymore, same applies from the frontend potentially

same argument is probably valid for the passkey setup